src/java/org/apache/cassandra/io/tries/Walker.java

```diff
@@ -83,7 +86,7 @@ protected final void go(long position)
         if (curOffset < 0 || curOffset >= buf.limit())
         {
             BufferHolder currentBh = bh;
-            bh = source.rebuffer(PageAware.pageStart(position));
+            bh = source.rebuffer(pageStart(position));
```

Comment:
Ugh. This is not as trivial as I hoped; now I remember why I was too lazy to test the simple writer: it does not enforce `Walker`'s requirement that no node should span a page boundary.

The test can still work if we memory-map the file in the simple writer case (with a comment why this is necessary).